File: Kotlin/k-closest-points-to-origin.kt
class Solution {
    fun kClosest(points: Array<IntArray>, k: Int): Array<IntArray> {
        points.sortWith(Comparator {
            a: IntArray, b: IntArray -> lengthSq(a) - lengthSq(b)
        })
        /*
        for (p in points) {
            val d= lengthSq(p)
            print("[${p[0]}, ${p[1]} | d=$d], ")
        }
        */
        val tmp = points.take(k)
        val result = Array<IntArray>(k) {i -> tmp[i]}
        return result
    }

    fun lengthSq(p: IntArray): Int {
        return p[0]*p[0] + p[1]*p[1]
    }
}

File: C++/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        sort(points.begin(), points.end(), [](vector<int> &a, vector<int> &b) {
            int da = a[0]*a[0] + a[1]*a[1];
            int db = b[0]*b[0] + b[1]*b[1];
            return da < db;
        });
        
        vector<vector<int>> ans;
        for (int i = 0; i < k; i++) {
            ans.push_back(points[i]);
        }
        return ans;
    }
};